### DISCUSSION TITLE
docs for openID connect usage

### DISCUSSION THREAD

**Discussion by matthdsm** [fastapi/fastapi#9073] [Questions]
### First check

* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

Hi,

Would it be possible to add a small manual on how to use openID connect in FastAPI? I know this has already been discussed in #12, but I'm having trouble getting it to work.
I've had mixed success using Authlib, but I'm not able to add this to the openapi spec, nor am I able to use the elegant `Depends` option to protect my API.

Any help would be much appreciated.

Thanks
M

**Comment by old-syniex** [comment]
I would like to join this question, i am totally loss

**Comment by davidmckayv** [comment]
I managed to find [this](https://docs.authlib.org/en/latest/client/fastapi.html), it was solid.  That said, I would also love to see docs on how to directly do this in FastAPI.  I liked @tiangolo 's [idea about social login then swapping with a FastAPI generated token](https://github.com/tiangolo/fastapi/issues/12#issuecomment-482544479), it would be outstanding to see this documented in the way he described.
